### PR TITLE
Fixed the non-working NRK connector

### PR DESCRIPTION
**Pull request by areee**
**Describe the changes you made**
I fixed the non-woking NRK connector.

**Additional context**
Näyttökuva 2020-7-10 kello 19 56 14

### CODE REVIEW

File: src/connectors/nrk-radio.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
-Connector.playerSelector = '.mini-player';
+Connector.playerSelector = '.app';
 
-Connector.artistTrackSelector = '.currently-playing .episode-title';
+Connector.artistTrackSelector = '.css-6752n5.e79qwfn0';
```
Comment (inverse) [inline src/connectors/nrk-radio.js:5]: This looks very brittle - like a new release of their site will cause things to break. Could we not use the `data-test="playingEpisodeDesc"` attribute to resolve this information?

File: src/connectors/nrk-radio.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 
-Connector.pauseButtonSelector = '.paused';
+Connector.pauseButtonSelector = '.css-1o5s8kn.e79qwfn0';
```
Comment (inverse) [inline src/connectors/nrk-radio.js:7]: Same with this one 

`data-test="playPauseMiniplayerMobile"` when `aria-label="Pause"`

**Comment by areee** [comment]
Thanks for your comments! I definitely agree! I'll try to enhance the selectors to more meaningful ones. :)

**Comment by areee** [comment]
I recently fixed those selectors to be less brittle. I hope they are what you were thinking about. :)

Besides, I added a filter to replace the '+' character with the '&' character. The NRK player seems to use '+' char when songs have multiple artists and that's why web-scrobbler does not recognize the played song.

File: src/connectors/nrk-radio.js [hunk 3]
```diff
@@ -1,7 +1,17 @@
...
 
-Connector.pauseButtonSelector = '.paused';
+Connector.artistTrackSelector = '[data-test="playingEpisodeDesc"]';
+
+Connector.pauseButtonSelector = '[data-test="playPauseMiniplayer"][aria-label="Pause"]';
+
+Connector.applyFilter(filter);
+
+function cleanUpArtist(artist) {
+	return artist.replace('+', '&');
```
Comment (alexesprit) [inline src/connectors/nrk-radio.js:16]: `String.replace` replaces a given substring one time only. So, if you have a "K-391 + Alan Walker + Julie Bergan + Seungri" string as an artist name, the output string will be "K-391 & Alan Walker + Julie Bergan + Seungri". This should be changed to keep consistent artist format.

You can use regular expressions to replace all matches.

**Comment by alexesprit** [comment]
Sorry for a delay.

The changes look good for me. I added a small change request, though. :)

File: src/connectors/nrk-radio.js (hunk 3 above)
Comment (inverse) [reply src/connectors/nrk-radio.js:16]: Thanks for explaining - I totally didn't realise that replace when passed string would only replace the first instance. Is that the same when passing regex without global flag? I guess so.

docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

Comment (alexesprit) [reply src/connectors/nrk-radio.js:16]: Yes, it's the same for expressions w/o a global flag.

https://jsfiddle.net/zmq548ov/

Comment (alexesprit) [reply src/connectors/nrk-radio.js:16]: Also, I think we could include spaces to make the filter even more robust, and to avoid unwanted replaces.

Comment (areee) [reply src/connectors/nrk-radio.js:16]: Hi! Thank you both for the information! I'll try to fix this by using the regex. :)

**Comment by areee** [comment]
Thanks for the regular expression tips, they were very helpful! 👍 Now I have made changes to the replacement filter feature. Is it what you were thinking about? :)

**Comment by alexesprit** [comment]
> Is it what you were thinking about? :)

Yes, it's perfect now. Thank your for contribution!

**Comment by hbhc** [comment]
I don't understand why scrobbling radio,nrk.no doesn't work for me at all. Any tips?

**Comment by areee** [comment]
The NRK connector is broken in the latest release (2.33.0) but it has already been fixed in the source code. You can install the extension from the source code or wait till the next release.

**Comment by alexesprit** [comment]
I'll publish a new release in few hours, it's better to wait for it.

**Comment by hbhc** [comment]
Cool. Thanks!

**Pull request merged**